app/controllers/application_controller.rb

```diff
@@ -26,66 +26,27 @@ def document_type
   end
 
   def current_format
-    document_types.fetch(params.fetch(:document_type, nil), nil)
+    document_types.detect { |format| format.document_type == document_type }
   end
 
   def formats_user_can_access
-    document_types.select { |_, v| policy(v.klass).index? }
+    document_types.select { |format| policy(format).index? }
   end
 
```

Comment:
Might not hurt to memoize this while we're at it too (not sure how much it's used though, if it's just in the drop down for the header maybe it's of less use).